### <font color = 'red'> Explore Hypothesis Testing

#### 1.2 Imports

import pandas as pd
import numpy as np
from scipy import stats

In [5]:
aqi = pd.read_csv('c4_epa_air_quality.csv',index_col = 0)
aqi.head()

date_local    state_name   county_name      city_name  \
0  2018-01-01       Arizona      Maricopa        Buckeye   
1  2018-01-01          Ohio       Belmont      Shadyside   
2  2018-01-01       Wyoming         Teton  Not in a city   
3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4  2018-01-01          Iowa          Polk     Des Moines   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1  Parts per million         0.263158    5  
2  Parts per million         0.111111    2  
3  Parts per million         0.300000    3  
4  Parts per million         0.215789    3

#### 1.3 Data Exploration

In [7]:
aqi.describe(include='all')

date_local  state_name  county_name      city_name local_site_name  \
count          260         260          260            260             257   
unique           1          52          149            190             253   
top     2018-01-01  California  Los Angeles  Not in a city         Kapolei   
freq           260          66           14             21               2   
mean           NaN         NaN          NaN            NaN             NaN   
std            NaN         NaN          NaN            NaN             NaN   
min            NaN         NaN          NaN            NaN             NaN   
25%            NaN         NaN          NaN            NaN             NaN   
50%            NaN         NaN          NaN            NaN             NaN   
75%            NaN         NaN          NaN            NaN             NaN   
max            NaN         NaN          NaN            NaN             NaN   

         parameter_name   units_of_measure  arithmetic_mean         aqi  
count               260                260       260.000000  260.000000  
unique                1                  1              NaN         NaN  
top     Carbon monoxide  Parts per million              NaN         NaN  
freq                260                260              NaN         NaN  
mean                NaN                NaN         0.403169    6.757692  
std                 NaN                NaN         0.317902    7.061707  
min                 NaN                NaN         0.000000    0.000000  
25%                 NaN                NaN         0.200000    2.000000  
50%                 NaN                NaN         0.276315    5.000000  
75%                 NaN                NaN         0.516009    9.000000  
max                 NaN                NaN         1.921053   50.000000

In [120]:
aqi['state_name'].value_counts().head()

state_name
California    66
Arizona       14
Ohio          12
Florida       12
Texas         10
Name: count, dtype: int64

#### 1.4 Statistical Tests

#### 1.4.1 Hypothesis 1 : ROA is considering a metropolitan-focused approach within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.a.

##### • H0:  There is no difference in the mean AQI between Los Angeles County and the rest of California. 
##### • HA: There is a difference in the mean AQI between Los Angeles County and the rest of California.

In [17]:
ca_la = aqi[aqi['county_name'] == 'Los Angeles']
ca_other = aqi[(aqi['state_name'] == 'California') & (aqi['county_name'] != 'Los Angeles')]

In [23]:
sig_level = 0.05

In [25]:
stats.ttest_ind(a=ca_la['aqi'] , b=ca_other['aqi'] , equal_var = False)

TtestResult(statistic=2.1107010796372014, pvalue=0.049839056842410995, df=17.08246830361151)

#### Since the p value = 0.049 is less than the significance level the null hypothesis is rejected. There is difference between aqi between LA county and rest of CA

#### 1.4.2 Hypothesis 2: With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?o?

##### • H0: The mean AQI of New York is greater than or equal to that of Ohio. 
##### • HA: The mean AQI of New York is below that of Ohio..

In [50]:
aqi_ny = aqi[aqi['state_name'] == 'New York']
aqi_ohio = aqi[aqi['state_name'] == 'Ohio']

In [105]:
meanny = aqi_ny['aqi'].mean()
meanny

2.5

In [107]:
meanohio = aqi_ohio['aqi'].mean()
meanohio

3.3333333333333335

In [91]:
tstat1,pvalue1 = stats.ttest_ind(a=aqi_ny['aqi'] , b=aqi_ohio['aqi'] , equal_var = False, alternative = 'less')
print(' Test Statistice value: ',tstat1,'\n','P value: ',pvalue1)

 Test Statistice value:  -2.025951038880333 
 P value:  0.03044650269193468


#### Since the p value = 0.0304 is less than the significance level the null hypothesis is rejected. The mean aqi of NY is below than that of ohio

#### 1.4.3 Hypothesis 3: A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy??

##### • H0: The mean AQI of Michigan is less than or equal to 10.
##### • HA: The mean AQI of Michigan is greater than 10..

In [89]:
aqi_michigan = aqi[aqi['state_name'] == 'Michigan']

In [98]:
mean = aqi_michigan['aqi'].mean()
mean

8.11111111111111

In [102]:
tstat2,pvalue2 = stats.ttest_1samp(a = aqi_michigan['aqi'], popmean = 10, alternative = 'greater')
print(' Test Statistice value: ',tstat2,'\n','P value: ',pvalue2)

 Test Statistice value:  -1.7395913343286131 
 P value:  0.9399405193140109


#### Since the p value = 0.939 is greater the null hypothesis is accepted. Therefore the mean aqi of michigan is less than 10 or equal to 10. Michigan will not be affected by the new policy